src/supermarket/app/workers/fieri_notify_worker.rb

```diff
@@ -21,9 +21,23 @@ def perform(cookbook_version_id)
     data = {
       'cookbook_name' => cookbook_version.name,
       'cookbook_version' => cookbook_version.version,
-      'cookbook_artifact_url' => cookbook_version.tarball.url
+      'cookbook_artifact_url' => cookbook_artifact_url(cookbook_version)
     }
 
     response = Net::HTTP.post_form(uri, data)
   end
+
+ def cookbook_artifact_url(cookbook_version)
+    if s3_configured?
+      cookbook_version.tarball.url
+    else
+      "#{Supermarket::Host.full_url}#{cookbook_version.tarball.url}"
+    end
+  end
+
+  def s3_configured?
+    %w(S3_BUCKET S3_ACCESS_KEY_ID S3_SECRET_ACCESS_KEY).all? do |key|
+      ENV[key].present?
```

Comment:
I _think_ we can ask `Paperclip` for [its storage option](https://github.com/chef/supermarket/blob/master/src/supermarket/config/initializers/paperclip.rb#L6-L16) somehow so that we only rely on the environment variables during initialization.